xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecFFmpeg.cpp

```diff
@@ -336,7 +336,16 @@ bool CDVDVideoCodecFFmpeg::Open(CDVDStreamInfo &hints, CDVDCodecOptions &options
   m_processInfo.SetSwDeinterlacingMethods();
   m_processInfo.SetVideoInterlaced(false);
 
-  pCodec = avcodec_find_decoder(hints.codec);
+  // libdav1d av1 sw decoding is implemented as a separate decoder
+  // in ffmpeg which is always found first when calling `avcodec_find_decoder`.
+  // To get hwaccels we look for decoders registered for `av1`.
+  // The decoder state check is needed to succesfully fallback to sw decoding if
+  // necessary.
+  if (hints.codec == AV_CODEC_ID_AV1 && m_decoderState != STATE_HW_FAILED)
+    pCodec = avcodec_find_decoder_by_name("av1");
```

Comment:
This seems strange to have to workaround this. Are you sure the dav1d decoder actually gets selected here? The hwaccell should get setup in the `get_format` callback. Otherwise I would expect this to be the same behaviour for x264, etc software decoders.

Reply:
As described in the comment:
The dav1d decoder is a standalone decoder in ffmpegs decoder list, which appears to be inserted int he codec list before the decoder named `av1` (which is libaom/hwaccell).
If you look at `avcodec_find_decoder` it just looks at the `codec_id` and returns the first possible decoder which in turn will always be dav1d as soon as it's enabled in ffmpeg.

This is not true for h264 and hevc, because the x264/x265 decoders are not pushed into the decoder list as standalone decoders.
You can also see this when you use the command `ffmpeg -decoders`:

```
 V..... libdav1d             dav1d AV1 decoder by VideoLAN (codec av1)
 V....D av1                  Alliance for Open Media AV1
VFS..D h264                 H.264 / AVC / MPEG-4 AVC / MPEG-4 part 10
 V..... h264_v4l2m2m         V4L2 mem2mem H.264 decoder wrapper (codec h264)
 VFS..D hevc                 HEVC (High Efficiency Video Coding)
 V..... hevc_v4l2m2m         V4L2 mem2mem HEVC decoder wrapper (codec hevc)
```

Seems the reason is that there are 2 possible sw decoders dav1d and libAOM and the hwaccell is bound to "av1".
Other projects like mpv select codecs by iterating over codecs with hwaccells so they deal that stuff differently, but honesty I don't want to rework our whole codec selection code, hence I came up with this.